cranelift-codegen/src/regalloc/liverange.rs

```diff
@@ -190,6 +190,22 @@ impl<'a, PO: ProgramOrder> bforest::Comparator<Ebb> for Cmp<'a, PO> {
     }
 }
 
+/// A simple helper macro to make comparisons more natural to read.
+macro_rules! cmp {
+    ($order:ident, $a:ident > $b:expr) => {
+        $order.cmp($a, $b) == Ordering::Greater
+    };
+    ($order:ident, $a:ident >= $b:expr) => {
+        $order.cmp($a, $b) != Ordering::Less
+    };
+    ($order:ident, $a:ident < $b:expr) => {
+        $order.cmp($a, $b) == Ordering::Less
+    };
+    ($order:ident, $a:ident <= $b:expr) => {
+        $order.cmp($a, $b) != Ordering::Greater
+    };
+}
```

Comment:
What about just implementing `Ord` and `Eq` on `ProgramOrder` or `ExpandedProgramPoint` or wherever it is needed so that the `<`, `<=`, `>`, etc. are overloaded and we can just use the operators directly instead of a macro or `order.cmp...`? It may require adding a method to `ProgramOrder` but the following seems more clear to me: `order.of(to) >= order.of(self.def_begin)`.

Reply:
Great suggestion; I'll open an issue to discuss this, since I'd like to make sure this is something we want first.